corehq/apps/users/views/utils.py

```diff
@@ -115,58 +116,84 @@ def log_commcare_user_locations_changes(request, user, old_location_id, old_assi
         )
 
 
-def _get_location_ids_with_single_user(domain, location_ids, user_id):
-    # Remove the user's ID, as we want to see which other CommCare users share any of the given location_ids
-    user_query = (
+def _flatten_list(list_2d):
+    flat_list = []
+    # Iterate through the outer list
+    for element in list_2d:
+        if isinstance(element, list):
+            # If the element is of type list, iterate through the sublist
+            for item in element:
```

Comment:
Do we expect a maximum depth of 2 for list? i.e
```
python["key1": "val", "key2": ["sub_key1": "subval1", "sub_key2": "subval2"]]
```

Reply:
Also, isn't there other flatten methods that we already use? This feels like a non-unique issue that might have been handled before in the codebase?

Reply:
The list will either have a depth of one or two. If only a single user is assigned to a location, it'll have a depth of one. Otherwise, if more than one user is assigned to a location, then the depth will be two.

Suprisingly, I couldn't find anything for flattening lists on the HQ codebase. In that case, at least we have something now. :smile:

Reply:
Nice! In that case, would it not make sense to put in somewhere in [this file](https://github.com/dimagi/commcare-hq/blob/11fcc3aafc1fb421073fd429d0fe35de13f94b27/corehq/apps/api/es.py)?

Reply:
Should we have it inside the ES file? As it's not specifically for ES, but can be used more generically as well. How about somewhere like [here](https://github.com/dimagi/commcare-hq/blob/11fcc3aafc1fb421073fd429d0fe35de13f94b27/corehq/apps/api/util.py)?

Reply:
I was thinking about this too - it's especially frustrating because elasticsearch actually does return values uniformly so they are always lists, we just unwrap single-element lists for backwards compatability:
https://github.com/dimagi/commcare-hq/blob/d114ccf2076a1c1003051c693e6ec6339cc80fcc/corehq/apps/es/utils.py#L56-L62

That only happens if you use `query.fields()` (which is deprecated) instead of `query.source()`, but unfortunately, `values_list` does use [`query.fields()`](https://github.com/dimagi/commcare-hq/blob/d114ccf2076a1c1003051c693e6ec6339cc80fcc/corehq/apps/es/es_query.py#L461-L461).  It'd be nice to switch this to the non-deprecated behavior, as that'd mean these `element`s are consistently lists.